components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment-store.js

```
@@ -8,27 +8,27 @@ export class AssignmentStore {
 	}
 
 	fetchAssignment(href, token) {
-
 		let promise = this._assignments.get(href);
 		if (!promise) {
-			promise = new Promise(async(resolve) => {
+			promise = new Promise((resolve, reject) => {

```

Cleaning up promise handling so that errors are not swallowed.

---

Perhaps just do this:

```
async function doFetch() {

    const assignment = new Assignment(href, token);
    await assignment.fetch();
    return assignment;
}

promise = doFetch();
```

So you can keep async-await

---

Yeh - that would be cleaner. 

If we go with the `ObjectStore` implementation with the observable map I'll be removing this anyway. But the `ObjectStore.fetch` has also ended up with some `then()` handling which could be refactored to use `await` too.